### ISSUE TITLE
Grad-CAM: obtaining classification score before softmax

### ISSUE THREAD

**Issue by sravan953** [keras-team/keras-io#370]
Hello,

The original Grad-CAM paper ([link](https://arxiv.org/pdf/1610.02391.pdf)) mentions obtaining the gradients of the class score y^c before softmax. However, the Keras implementation ([link](https://github.com/keras-team/keras-io/blob/master/examples/vision/grad_cam.py)) does not seem to do this (added in commit e005c9014d19f3dffb0ca983b20eb5a2c3864a3a)? Is this a bug or am I interpreting it incorrectly?

Thank you!

**Comment by YassineYousfi** [comment]
Line 97 does dy^c/dA then line 101 computes the global average pooling in equation (1) of the paper.

**Comment by YassineYousfi** [comment]
Agree that the derivatives are taken after softmax here unlike the original grad-cam. Although I haven't seen visual differences between cams produced by before/after softmax.
PS. Can be easily fixed by setting the last layer's activation to None.

**Comment by sravan953** [comment]
The only concern is that the final heatmap might not be entirely faithful since it has been distorted by a non-linear activation function?

**Comment by sachinprasadhs** [comment]
@sravan953 , Could you please verify the commit https://github.com/keras-team/keras-io/commit/543ece4dbf0cb540796894eddca447fcecd971a0 which addressed this issue.

If you are satisfied with the above commit, feel free to close this issue. Thanks!
